### ISSUE TITLE
Panic in transmute.rs:33

### ISSUE THREAD

**Issue by afck** [Manishearth/rust-clippy#700]
Clippy panics in transmute.rs:33 with today's Rust nightly when I run it on https://github.com/afck/routing/commit/28da2d60af947c7924cc5f4c3fed09b756d26a7e:
```
$ RUST_BACKTRACE=1 cargo test --no-run --features clippy
   Compiling routing v0.6.3 (file:///home/andreas/git/routing)
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
thread 'rustc' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/andreas/.cargo/registry/src/github.com-88ac128001ac3a9a/clippy-0.0.43/src/transmute.rs:33
stack backtrace:
   1:     0x7f30eab03bc0 - sys::backtrace::tracing::imp::write::hb39adb9cc1dae4a9Nyu
   2:     0x7f30eab0cf2b - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.43683
   3:     0x7f30eab0ca83 - panicking::default_handler::hced6c0d85eb2cb83bez
   4:     0x7f30eaad55fc - sys_common::unwind::begin_unwind_inner::hafb014b2dc8828386mt
   5:     0x7f30eaad6098 - sys_common::unwind::begin_unwind_fmt::hd671b811c60b4ff4cmt
   6:     0x7f30eab01691 - rust_begin_unwind
   7:     0x7f30eab5415f - panicking::panic_fmt::h38cc775b27fedfe3yZL
   8:     0x7f30eab54342 - panicking::panic_bounds_check::h74dcd407bdd3e0b8EYL
   9:     0x7f30dc9cb19e - transmute::UselessTransmute.LateLintPass::check_expr::hc601156b2617405co2k
                        at /home/andreas/.cargo/registry/src/github.com-88ac128001ac3a9a/clippy-0.0.43/src/transmute.rs:33
  10:     0x7f30e7fa314a - lint::context::LintContext::with_lint_attrs::h380019028915481644
  11:     0x7f30e7fa47cb - lint::context::LateContext<'a, 'tcx>.hir_visit..Visitor<'v>::visit_stmt::h08e6ba19683690b35cv
  12:     0x7f30e7fa3eae - lint::context::LateContext<'a, 'tcx>.hir_visit..Visitor<'v>::visit_block::h432b9e92e3858b0d9lv
  13:     0x7f30e7f9eeda - lint::context::LateContext<'a, 'tcx>.hir_visit..Visitor<'v>::visit_fn::h7b178fed5cfd4cbdOdv
  14:     0x7f30e7f9adee - lint::context::LateContext<'a, 'tcx>.hir_visit..Visitor<'v>::visit_item::h91102ff40201017eW8u
  15:     0x7f30e7f9f17b - lint::context::LateContext<'a, 'tcx>.hir_visit..Visitor<'v>::visit_mod::h934956317d2c89a9nkv
  16:     0x7f30e7fbffea - lint::context::check_crate::ha9a780d15a42d7af15v
  17:     0x7f30eb027de3 - driver::phase_3_run_analysis_passes::_$u7b$$u7b$closure$u7d$$u7d$::closure.28414
  18:     0x7f30eb0221f5 - middle::ty::context::ctxt<'tcx>::create_and_enter::h2830587858154539181
  19:     0x7f30eb01ed4f - driver::phase_3_run_analysis_passes::h15248650410224887609
  20:     0x7f30eaff05c5 - driver::compile_input::h952463538ef70172Bca
  21:     0x7f30eafe0917 - run_compiler::hcb7308fe894426e8HOc
  22:     0x7f30eafde0c1 - sys_common::unwind::try::try_fn::h16635096110630877533
  23:     0x7f30eab0161b - __rust_try
  24:     0x7f30eaaf9bcd - sys_common::unwind::inner_try::hacd4193e6d9e10788jt
  25:     0x7f30eafde910 - boxed::F.FnBox<A>::call_box::h6111647763662046071
  26:     0x7f30eab0b4f9 - sys::thread::Thread::new::thread_start::h42fba1e2d4e7a4bevby
  27:     0x7f30e32920a3 - start_thread
  28:     0x7f30ea78187c - clone
  29:                0x0 - <unknown>

Build failed, waiting for other jobs to finish...
Could not compile `routing`.

To learn more, run the command again with --verbose.
```
Here is an example on Travis: https://travis-ci.org/maidsafe/routing/jobs/111015551#L367

**Comment by llogiq** [comment]
Possibly the `&args[0]`, but that would mean we somehow call transmute without arguments!?

**Comment by mcarton** [comment]
Looks like it fail at:
```rust
fn main() {
    init();
    core(); // HERE
}
```

**Comment by Manishearth** [comment]
match_def_path doesn't check that the lengths are the same :/

**Comment by mcarton** [comment]
@afck it should be fixed in 0.0.44.

**Comment by afck** [comment]
It works! Thank you! :+1:
